app/controllers/edit_merchant_details/post-edit-controller.js

```diff
@@ -16,21 +16,39 @@ const ADDRESS_POSTCODE = 'address-postcode'
 const ADDRESS_COUNTRY = 'address-country'
 const MERCHANT_EMAIL = 'merchant-email'
 
+const trimField = (key, store) => lodash.get(store, key, '').trim()
+
 module.exports = (req, res) => {
   const correlationId = lodash.get(req, 'correlationId')
   const externalServiceId = req.service.externalId
   const hasDirectDebitGatewayAccount = lodash.get(req, 'service.hasDirectDebitGatewayAccount') || lodash.get(req, 'service.hasCardAndDirectDebitGatewayAccount')
+
+  const fields = [
+    MERCHANT_NAME,
+    TELEPHONE_NUMBER,
+    ADDRESS_LINE1,
+    ADDRESS_LINE2,
+    ADDRESS_CITY,
+    ADDRESS_POSTCODE,
+    ADDRESS_COUNTRY,
+    MERCHANT_EMAIL
+  ]
+  const formFields = fields.reduce((form, field) => {
+    form[field] = trimField(field, req.body)
+    return form
+  }, {})
+
   const reqMerchantDetails = {
-    name: req.body[MERCHANT_NAME],
-    telephone_number: req.body[TELEPHONE_NUMBER] ? req.body[TELEPHONE_NUMBER].replace(/\s/g, '') : req.body[TELEPHONE_NUMBER],
-    email: req.body[MERCHANT_EMAIL],
-    address_line1: req.body[ADDRESS_LINE1],
-    address_line2: req.body[ADDRESS_LINE2],
-    address_city: req.body[ADDRESS_CITY],
-    address_postcode: req.body[ADDRESS_POSTCODE],
-    address_country: req.body[ADDRESS_COUNTRY]
+    name: formFields[MERCHANT_NAME],
+    telephone_number: formFields[TELEPHONE_NUMBER].replace(/\s/g, ''),
+    email: formFields[MERCHANT_EMAIL],
+    address_line1: formFields[ADDRESS_LINE1],
+    address_line2: formFields[ADDRESS_LINE2],
+    address_city: formFields[ADDRESS_CITY],
+    address_postcode: formFields[ADDRESS_POSTCODE],
+    address_country: formFields[ADDRESS_COUNTRY]
```

Comment:
I wish it was obvious that these keys need to match what adminusers expects in the `PATCH`…

Reply:
hmm maybe there is a way to make this better. I'll look into this in the cleanup at the end